crates/nu-cli/src/repl.rs

```diff
@@ -445,27 +512,41 @@ pub fn evaluate_repl(
                         &s,
                         engine_state,
                         cmd_duration,
-                        stack,
+                        &mut unique_stack,
                         &mut line_editor,
                     )?;
                 }
 
                 if shell_integration {
-                    do_shell_integration_finalize_command(hostname, engine_state, stack)?;
+                    do_shell_integration_finalize_command(
+                        hostname,
+                        engine_state,
+                        &mut unique_stack,
+                    )?;
                 }
 
                 flush_engine_state_repl_buffer(engine_state, &mut line_editor);
             }
             Ok(Signal::CtrlC) => {
                 // `Reedline` clears the line content. New prompt is shown
                 if shell_integration {
-                    run_ansi_sequence(&get_command_finished_marker(stack, engine_state))?;
+                    let stack = Arc::new(unique_stack);
```

Comment:
`get_command_finished_marker` takes `&Stack`, no need to create a new `Arc` here.